### PR TITLE
Fix instanceof check and cast with pattern matching (#85)

### PR DESCRIPTION
**Pull request by dirk-olmes**
related to #85

### CODE REVIEW

File: engine-dmn/engine/src/main/java/org/operaton/bpm/dmn/engine/impl/type/DateDataTypeTransformer.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ public TypedValue transform(Object value) throws IllegalArgumentException {
       Date date = transformString(stringValue);
       return Variables.dateValue(date);
 
-    } if (value instanceof ZonedDateTime zonedDateTimeValue) {
+    } else if (value instanceof ZonedDateTime zonedDateTimeValue) {
       Instant instant = (zonedDateTimeValue).toInstant();
```
Comment (kthoms) [inline engine-dmn/engine/src/main/java/org/operaton/bpm/dmn/engine/impl/type/DateDataTypeTransformer.java:60]:
```suggestion
      Instant instant = zonedDateTimeValue.toInstant();
```

**Review by kthoms — CHANGES_REQUESTED** [review]

File: engine-cdi/core/src/main/java/org/operaton/bpm/engine/cdi/impl/annotation/StartProcessInterceptor.java [hunk 2]
```diff
@@ -63,8 +63,8 @@ public Object invoke(InvocationContext ctx) throws Exception {
       return result;
     } catch (InvocationTargetException e) {
       Throwable cause = e.getCause();
-      if(cause != null && cause instanceof Exception) {
-        throw (Exception) cause;
+      if (cause != null && cause instanceof Exception exception) {
```
Comment (javahippie) [inline engine-cdi/core/src/main/java/org/operaton/bpm/engine/cdi/impl/annotation/StartProcessInterceptor.java:66]: Just a question on how to proceed here: A user commented on a commit some time earlier, that `instanceof` is null safe. If the passed parameter is null, it is not an instance of `Exception` and the check is `false`, skipping the body. Is this something we should already improve here?

Comment (javahippie) [reply engine-cdi/core/src/main/java/org/operaton/bpm/engine/cdi/impl/annotation/StartProcessInterceptor.java:66]: See https://docs.oracle.com/javase/specs/jls/se17/html/jls-15.html#jls-15.20.2

> At run time, the result of the type comparison operator is determined as follows:
>
 >   If the value of the RelationalExpression is the null reference ([§4.1](https://docs.oracle.com/javase/specs/jls/se17/html/jls-4.html#jls-4.1)), then the result is false.

Comment (kthoms) [reply engine-cdi/core/src/main/java/org/operaton/bpm/engine/cdi/impl/annotation/StartProcessInterceptor.java:66]: the null check is obsolete then

**Review by kthoms — APPROVED** [review]

**Comment by kthoms** [comment]
Sorry @dirk-olmes I did not recognize that the findings have already been adressed for a long time. Just ping in a comment. Did you know that you could (re-)request a review when clicking on one of the reviewers in top right?

**Pull request merged**